citadel_workspace_service/src/kernel/mod.rs

```diff
@@ -539,14 +524,24 @@ mod tests {
                         bincode2::deserialize(&*next_packet)?;
                     info!(target: "citadel","{response_message_packet:?}");
 
-                    if let InternalServiceResponse::MessageReceived {
-                        cid,
-                        message: _,
-                        peer_cid: _,
-                    } = response_message_packet
-                    {
+                    if let InternalServiceResponse::MessageSent { cid } = response_message_packet {
                         info!(target:"citadel", "Message {cid}");
-                        Ok(())
+                        let next_packet = stream.next().await.unwrap()?;
+                        let response_message_packet: InternalServiceResponse =
+                            bincode2::deserialize(&*next_packet)?;
+                        if let InternalServiceResponse::MessageReceived {
+                            message,
+                            cid,
+                            peer_cid: _,
+                        } = response_message_packet
+                        {
+                            println!("{message:?}");
+                            assert_eq!("pong", String::from_utf8_lossy(&*message));
```

Comment:
Convert the str into a SecBuffer, and compare the SecBuffer's
```suggestion
                            assert_eq!(SecBuffer::from("pong"), message);
```